document `download_instructions` enhancements in EasyBuild 5.0 (#243)
easybuilders/easybuild-docs: docs/easybuild-v5/enhancements.md

- [Revamp of easyconfig parameter `modextrapaths`][modextrapaths-revamp]
- [Detect Fortran `.mod` files in `GCCcore` installations][mod-files]
- [Let `ConfigureMake` generic easyblock error out on unknown `configure` options][configuremake-unknown-configure-options]

> - [Require `download_instructions` to be specified][require_download_instructions]

Reviewer: ```suggestion
- [Require `download_instructions` for non-public sources][require_download_instructions]
```

    [override]
    trace=0
    ```

---

## Require `download_instructions` to be specified {: require_download_instructions }

[`download_instructions`][download_instructions] is used in easyconfigs to specify instructions, or information, on

> how to obtain sources that are not directly downloadable. In EasyBuild 5, we require that easyconfigs specify these

Reviewer 2: That's not actually entirely true, we only require this in the easyconfigs test suite, *not* in EasyBuild itself...

That's big difference, so we should reword this imho...

> repository to require that all easyconfig PR specify these (see [easybuild-easyconfigs PR #19881](https://github.com/easybuilders/easybuild-easyconfigs/pull/19881)).

Reviewer 2: ```suggestion
repository to require that all easyconfigs for which sources can not be downloaded automatically must specify these
(see [easybuild-easyconfigs PR #19881](https://github.com/easybuilders/easybuild-easyconfigs/pull/19881)).

If the software uses the `.mod` extension for a different type of file then the EasyConfig variable
`skip_mod_files_sanity_check` can be set to mark that this use is expected.

---

## Let `ConfigureMake` generic easyblock error out on unrecognized `configure` options { : #configuremake-unrecognized-configure-options }

*(more info soon)*

---

> ## Require `download_instructions` to be specified {: require_download_instructions }

Reviewer: ```suggestion
## Require `download_instructions` for non-public sources {: require_download_instructions }
```

> [`download_instructions`][download_instructions] is used in easyconfigs to specify instructions, or information, on

Reviewer: ```suggestion
[`download_instructions`][download_instructions] is a new easyconfig parameter
in EasyBuild 5.0 used to specify instructions, or information, on
```

> how to obtain sources that are not directly downloadable. In EasyBuild 5, we've updated the testsuite for the `easybuild-easyconfigs`
> repository to require that all easyconfigs for which sources can not be downloaded automatically must specify these

Reviewer: ```suggestion
how to obtain software sources that are not directly downloadable. This is
usually the case for licensed software without public sources. In EasyBuild 5,
we've updated the testsuite for our `easybuild-easyconfigs` repository to
require that all its easyconfigs must specify the download instructions
of any non-public sources in them
```

If the software uses the `.mod` extension for a different type of file then the EasyConfig variable
`skip_mod_files_sanity_check` can be set to mark that this use is expected.

---

## Let `ConfigureMake` generic easyblock error out on unrecognized `configure` options { : #configuremake-unrecognized-configure-options }

*(more info soon)*

---

## Require `download_instructions` for non-public sources {: require_download_instructions }

[`download_instructions`][download_instructions] is a new easyconfig parameter
in EasyBuild 5.0 used to specify instructions, or information, on
how to obtain software sources that are not directly downloadable. This is

> usually the case for licensed software without public sources. In EasyBuild 5,
> we've updated the testsuite for our `easybuild-easyconfigs` repository to
> require that all its easyconfigs must specify the download instructions
> of any non-public sources in them
> (see [easybuild-easyconfigs PR #19881](https://github.com/easybuilders/easybuild-easyconfigs/pull/19881)).

Reviewer: ```suggestion
usually the case for licensed software without public sources.

We've updated the testsuite for our `easybuild-easyconfigs` repository to
require that all its easyconfigs must specify the download instructions
of any non-public sources in them
(see [easybuild-easyconfigs PR #19881](https://github.com/easybuilders/easybuild-easyconfigs/pull/19881)).
```
